app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java

```diff
@@ -137,4 +152,36 @@ private int aspectRatio(int width, int height){
         return (Math.abs(previewRatio - RATIO_4_3_VALUE) <= Math.abs(previewRatio - RATIO_16_9_VALUE))
                 ? AspectRatio.RATIO_4_3 : AspectRatio.RATIO_16_9;
     }
+
+    /**
+     * Get the horizontal and vertical field of view of the back-facing amera
+     * @return the horizontal and vertical field of view in degrees
+     */
+    public Pair<Float, Float> getFieldOfView(){
+        //Create camera manager
+        CameraManager cameraManager = (CameraManager) context.getSystemService(Context.CAMERA_SERVICE);
+        //Initialize horiontal and vertical fov
+        double horizontalAngle = 0;
+        double verticalAngle = 0;
+        try {
+            //Go through every camera to get the back-facing camera
```

Comment:
I am not entirely sure if this works, but have you tried using something like:

public boolean isRearCameraAvailable(Context context) {
    return context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA);
}

to check if a back-facing camera is available?

Reply:
Added PackageManager to check if the device has a camera